### DISCUSSION TITLE
Nano ELS H5

### DISCUSSION THREAD

**Discussion by Festivejelly** [kachurovskiy/nanoels#232] [General]
Loving the new H5 you added. Looking on getting one made. Touch screen is quite expensive but I think it will be a good choice over having lots of buttons.

Are you likely to ever sell these in your shop? If not ill order the PCB's, never done it before but the instructions seem clear.

Your H4 was already a game changer and I think the H5 with the extra inputs etc is going to raise the bar even higher!

**Comment by kachurovskiy** [comment]
Hi @Festivejelly, thank you. I don't plan to sell anything myself. Have high hopes for H5 as well! 😀

**Reply by Festivejelly** [reply]
I've ordered the boards now and the components so hopefully around mid January I can get it all up and running.

I like the new board design and the dedicated power for the controller. That will make a difference for the encoder and stepper drivers.

**Comment by duebelpapst** [comment]
Hi Maxim,
nice to hear that you still working on the Nanoels.
Like Festivejelly I never orderd PCBs in China till now, I am looking forward to the arrival of the PCBs.

Regards,
Stefan

**Comment by demsarmatjaz** [comment]
Hi Maxim,
I succeded to get the H5 running after some problems which were actually not the problems with the code but with my knowledge :-).
Now I am in the phase of testing the Nano ELS H5 version, currently on my desk, but later, when it will be stable enough I will transfer everything to my Bernardo Proficenter 700 BQV. Otherwise, I am very impressed with the concept of the device, but I am noticing some problems in operation that I am sure, will be solved. At first, I thought that my encoder was not working, but later it turned out that the problem was most likely just my lack of knowledge of the STM32 Arduino environment and the operating logic of the Nano ELS. Since I am not familiar with the H4 version, it would be nice to have a list of keyboard commands with descriptions of functions and setting procedures for individual operations. I know that there are comments in the code for keyboard scancodes (from line 115 and below, as I remember) but that is not enough for someone who didn't use the H4 in practice before. I tried to follow the instructions for the H4 version, but certain settings did not work for me or I approached them in the wrong way, so some detailed user manual would be welcome. Now to the problems:
- I can follow the encoder angle on the screen without error, but it does not show me the revolutions - rpm. The rpm display is always 0,
- Another problem is the operation of the keyboard. Sometimes I have to press a key several times for the interface to react, sometimes I press once but it repeats the input several times. The keyboard is industrial foil type, made for harsh environment and when I connect it to the computer it works 100%,
- It also happens that when I try to change, let's say, the X position by pressing corresponding arrow key on the keyboard, the stepper motor moves equivalently to a one milimeter of movement, but sometimes pressing an arrow key gets stuck somehow and the motor continue running until I stop it by pressing again the same arrow key, which sometimes works after the first press, and sometimes it takes several presses to stop it.

I will be thankfull for any suggestions and help.

**Reply by kachurovskiy** [reply]
Hi @demsarmatjaz, thanks a lot for the kind words and being the early adopter of H5. 

> it would be nice to have a list of keyboard commands with descriptions of functions and setting procedures for individual operations. I know that there are comments in the code for keyboard scancodes (from line 115 and below, as I remember) but that is not enough for someone who didn't use the H4 in practice before.

Yes, it's definitely on me to at least describe which buttons correspond to which actions.

To fix RPM delete these lines - https://github.com/kachurovskiy/nanoels/blob/main/h5/h5.ino#L536-L539 - or just wait for the next version.

Keyboard support definitely needs more work, currently I only store 1 incoming code for processing, it can be overridden before it's processed - https://github.com/kachurovskiy/nanoels/blob/main/h5/h5.ino#L492-L523

Some keyboards don't work because they expect an initialization sequence sent and the code doesn't do that yet.

**Reply by kachurovskiy** [reply]
@demsarmatjaz all the issues you mentioned should now be resolved in the latest version, please let me know if not.

**Comment by demsarmatjaz** [comment]
Hi Maxim,

Perfect, thanks a lot. I checked the new version, and everything works flawlessly. I'm looking forward to new updates.

Kind Regards,
Matjaz D.

**Comment by Viktor1972-avi** [comment]
Максим добрый день. Несколько вопросов по H5:
1. Возможно подключенияя джойстика и ручного генератора импульсов?
2. Будет поддержка работы с G кодом?
3. WiFi для каких целей можно будет использовать?

**Reply by kachurovskiy** [reply]
Добрый день, Виктор.

1. Порт для резисторного 3-осевого джойстика предусмотрен но поддержка в коде пока не реализована. РГИ на каждую ось отдельно есть.
2. Планируется через WiFi, надеюсь получится. Классическая поддержка через серийный интерфейс видимо перестала работать из-за конфликта между несколькими серийными интерфейсами.
3. См. пункт 2 :)

**Reply by Viktor1972-avi** [reply]
Собираюсь плату заказывать, из за проблеммы конфликта между несколькими серийными интерфейсами, ее конструктив не изменится? Т.е можно заказывать?

**Reply by kachurovskiy** [reply]
На данный момент плату менять не планирую
